holopy/scattering/calculations.py

```diff
@@ -23,20 +23,21 @@
 """
 
 from holopy.core.holopy_object import SerializableMetaclass
-from holopy.core import Optics, Schema, Image, VectorGrid, interpret_args
+from holopy.core.metadata import vector, Image, optical_parameters
 from holopy.core.tools import dict_without, is_none
 from holopy.scattering import Mie, Multisphere, Sphere, Spheres
 from holopy.scattering.theory import dda
 from holopy.scattering.errors import AutoTheoryFailed, MissingParameter
 
 import numpy as np
+from xarray.ufuncs import square
 
 def check_schema(schema):
```

Comment:
The way I have structured the new code I believe what will happen if you are missing one of the parameters is you will get an error about a missing parameter to one of the internal scattering functions. Which is probably not ideal as a user experience. I think what might make sense is to do some kind of checking in the optical_parameters function since that is used in all the right places and is the place that unifies data from the schema and function arguments.

Currently it would be a little difficult because I purposefully structured the optical_parameters function to gracefully handle whatever data you ask it for. However, I think there are only a small set off different cases (mostly whether you need polarization or not) so we could have two or three similar functions that do the checking and raise helpful errors.